src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx

```diff
@@ -0,0 +1,70 @@
+With the release of v1 amplify-flutter now supports Web and Desktop for Auth, API, Analytics, and Storage use cases.  Developers can now build cross-platform Flutter apps with Amplify that target iOS, Android, Web, and Desktop (macOS, Windows, Linux) using a single codebase.
+
+We have re-written our libraries in Dart. In some places, we have made breaking changes to improve ergonomics or enable features that had been missing from the v0 implementations.
+
+## Prerequisites
+
+* A Flutter application targeting Flutter SDK >= 3.0.0 with Amplify libraries integrated
+  
+  The following are also required, depending on which platforms you are targeting:
+
+  * An iOS configuration targeting at least iOS 13.0 and XCode version >=13.2
+  * An Android configuration targeting at least Android API level 24 (Android 7.0) or above
+  * Any browser supported by Flutter for Web (you can check the list of supported browsers [here](https://docs.flutter.dev/development/platform-integration/web/faq#which-web-browsers-are-supported-by-flutter))
+  * Any Windows OS meeting Flutter minimums
+  * macOS version 10.15 or higher
+  * Any Ubuntu Linux distribution meeting Flutter minimums
+  * For a full example please follow the [project setup walkthrough](/lib/project-setup/create-application)
+
+## Auth
+
+- Changes to API options
+- Changes to exception handling
+- Changes to nextStep types
+- getPlugin for Cognito-specific behavior
+- Changes to Hub events
```

Comment:
See below for Auth category changes. I was going to leave this as a suggested change, but that doesn't work when there are code snippets.

<details>

- `fetchAuthSession` had several changes. See the example below for how to migrate.
  - `Amplify.Auth.getPlugin()` has been added to return plugin specific sub types. Casting to `CognitoAuthSession` is no longer needed when using this.
  - `getAWSCredentials` has been deprecated. AWS credentials will always be included in the response.
  - `fetchAuthSession` will no longer throw exceptions. Instead, indivdual getters of type `AuthResult` will throw if the underlying operation fails.
    - _Why?_ The previous behavior did not allow for partial successes in offline scenarios when user pool tokens were still valid, but AWS credentials were expired. See [amplify-flutter#760](https://github.com/aws-amplify/amplify-flutter/issues/760#issuecomment-1453559459) for more details.
  - `CognitoAuthSession.credentials`, `CognitoAuthSession.userPoolTokens`, `CognitoAuthSession.`, and ``CognitoAuthSession.` are deprecated and will be removed in a future release.

```dart
// v0
try {
  final session = await Amplify.Auth.fetchAuthSession(
    options: CognitoFetchAuthSessionOptions(getAWSCredentials: true),
  ) as CognitoAuthSession;

  final accessToken = session.userPoolTokens?.accessToken;
  safePrint('accessToken: $accessToken');

  final identityId = session.identityId;
  safePrint('identityId: $identityId');
} on AuthException catch (e) {
  // fetchAuthSession will fail if any of the underlying operations fail.
  safePrint('The operation failed: ${e.message}');
}
```

```dart
// v1
final session = await Amplify.Auth.getPlugin(
  AmplifyAuthCognito.pluginKey,
).fetchAuthSession();

// AuthResult.value will throw if the underlying operation failed.
try {
  final accessToken = session.userPoolTokensResult.value.accessToken.toJson();
  safePrint('accessToken: $accessToken');
} on SignedOutException {
  // the user is not signed in.
} on SessionExpiredException {
  // the users session has expired.
} on NetworkException {
  // the access and/or id token is expired but cannot be refreshed because the
  // user is offline.
}

// `AuthResult.valueOrNull` will return the identityId if present or null,
// but will never throw even if the underlying operation failed.
final identityId = session.identityIdResult.valueOrNull;
safePrint('identityId: $identityId');
```

- The next step returned from auth operations are now enums. See the example below for how to migrate for `Auth.signUp()`.

```dart
// v0
final result = await Amplify.Auth.signUp(/* ... */);
switch (result.nextStep?.signUpStep) {
  case 'CONFIRM_SIGN_UP':
    // prompt user to confirm sign in using nextStep.codeDeliveryDetails
    break;
  case 'DONE':
    // sign up is complete, user can be taken to an authenticated state.
    break;
}

// v1
final result = await Amplify.Auth.signUp(/* ... */);
switch (result.nextStep.signUpStep) {
  case AuthSignUpStep.confirmSignUp:
    // prompt user to confirm sign in using nextStep.codeDeliveryDetails
    break;
  case AuthSignUpStep.done:
    // sign up is complete, user can be taken to an authenticated state.
    break;
}
```

- Auth hub events are now have a specific sub type, `AuthHubEvent` and contain a `type` that is an enum. See the example below for how to migrate.

```dart
// v0
final hubSubscription = Amplify.Hub.listen([HubChannel.Auth], (hubEvent) {
  switch(hubEvent.eventName) {
    case 'SIGNED_IN':
      safePrint('User is signed in.');
      break;
    case 'SIGNED_OUT':
      safePrint('User is signed out.');
      break;
    case 'SESSION_EXPIRED':
      safePrint('The session has expired.');
      break;
    case 'USER_DELETED':
      safePrint('The user has been deleted.');
      break;
  }
});
```

```dart
// v1
final subscription = Amplify.Hub.listen(HubChannel.Auth, (AuthHubEvent event) {
  switch (event.type) {
    case AuthHubEventType.signedIn:
      safePrint('User is signed in.');
      break;
    case AuthHubEventType.signedOut:
      safePrint('User is signed out.');
      break;
    case AuthHubEventType.sessionExpired:
      safePrint('The session has expired.');
      break;
    case AuthHubEventType.userDeleted:
      safePrint('The user has been deleted.');
      break;
  }
});
```

- Cognito specific API options such as `CognitoSignInOptions` are deprecated in favor of `pluginOptions`. The classes will be removed in a future release. See below for an example of how to migrate for `Auth.signIn()`.

```dart
// v0
final result = await Amplify.Auth.signIn(
  username: username,
  password: password,
  options: const CognitoSignInOptions(
    clientMetadata: {'data': 'value'},
  ),
);

// v1
final result = await Amplify.Auth.signIn(
  username: username,
  password: password,
  options: const SignInOptions(
    pluginOptions: CognitoSignInPluginOptions(
      clientMetadata: {'data': 'value'},
    ),
  ),
);
```
</details>

Reply:
Just confirming these are the changes you already comitted?